File: ejercicios_csura/Semana01/soluciones/Ejercicio05.cs
using Semana01.soluciones;

namespace soluciones
{
    public class Ejercicio05 : IEjercicio
    {
        public void Ejecutar()
        {
            try
            {
                Console.WriteLine("Ingrese el primer número:");
                double a = double.Parse(Console.ReadLine().Trim());

                Console.WriteLine("Ingrese el segundo número:");
                double b = double.Parse(Console.ReadLine().Trim());

                Console.WriteLine($"La suma de {a} + {b} es: {Math.Round((a + b), 2)}");
            }
            catch
            {
                Console.WriteLine("Error: Debe ingresar un número válido.");
                return;
            }
        }
    }
}

File: ejercicios_csura/Semana03/soluciones/Ejercicio05.vb
Public Class Ejercicio05
    Implements IEjercicio

    Public Sub Ejecutar() Implements IEjercicio.Ejecutar
        Dim numero As Integer = 0
        Dim numeroValido As Boolean = False

        Do While Not numeroValido
            Try
                Console.WriteLine("Ingrese un número entero:")
                numero = CInt(Console.ReadLine().Trim())
            Catch
                Console.WriteLine($"Error: Ingrese un número entero válido.{vbCrLf}")
                Continue Do
            End Try

            numeroValido = True
        Loop

        Console.WriteLine($"{vbCrLf}El número {numero} es {If(numero Mod 2 = 0, "par", "impar")}.")
    End Sub
End Class
